app/assets/v2/js/status.js

```diff
@@ -21,7 +25,61 @@ $(document).ready(function() {
 
   // dropdown for usernames when @ is detected in the post
   $('#textarea').on('input', function(e) {
+    e.preventDefault();
     const lastWord = e.target.value.split(' ').pop();
+    const site = e.target.value.match(url_re);
+    const youtube = e.target.value.match(youtube_re);
+    const no_lb = e.originalEvent.inputType !== 'insertLineBreak';
+    
+    if (youtube !== null && youtube[1].length === 11 && no_lb) {
+      let videoId = youtube[1];
+      
+      if (embedded_resource !== youtube[0]) {
+        var apiKey = 'AIzaSyDi-EFpC2ntx9PnM_-oiJHk5zCY53KdIf0'; // TODO: add youtube API key to query titles
```

Comment:
Could we define this in [settings.py](https://github.com/gitcoinco/web/blob/master/app/app/settings.py) and make it such that the user can define it in their [.env](https://github.com/gitcoinco/web/blob/master/app/app/local.env
) file
the `YOUTUBE_API_KEY` would have to be passed from the backend -> which we can use here